### PR TITLE
memory_leak_after_nichotplug: Add new case

### PR DESCRIPTION
**Pull request by xiywang**
id: 1666145, 1688093

1. Do memory leak check after nic hotplug/hotunplug for windows
2. Do memory leak check after add/del vlan for linux

Signed-off-by: Xiyue Wang <xiywang@redhat.com>

### CODE REVIEW

**Comment by xiywang** [comment]
(1/2) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.my.memory_leak_after_nichotplug: ERROR: There might be memory leak after hotplug nic.\nMemory reduced 846 (262.88 s)
 (2/2) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.my.memory_leak_after_nichotplug: PASS (1206.74 s)

There's an existing bug of memory leak in RHEL guests after add/del vlan.
So the test result of RHEL is test failed.

File: qemu/tests/memory_leak_after_nichotplug.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    def cleanup_ovs_ports(netdst, ports):
+        if is_ovs_backend(netdst) is True:
+            ports = set(get_ovs_ports(netdst).splitlines()) - \
+                set(ports.splitlines())
+            for p in ports:
+                process.system("ovs-vsctl del-port %s %s" % (netdst, p))
+
+    netdst = params.get("netdst", "switch")
+    host_hw_interface = get_ovs_ports(netdst)
```
Comment (pezhang) [inline qemu/tests/memory_leak_after_nichotplug.py]: @xiywang Seems this is not a ovs backend only case. For bridge backend, test will fail:

ERROR: Command 'ovs-vsctl list-ports switch' failed.\nstdout: b''\nstderr: b'ovs-vsctl: unix:/var/run/openvswitch/db.sock: database connection failed (No such file or directory)\n'\nadditional_info: None

**Comment by xiywang** [comment]
@pezhang Updated. Please help review again. Thanks.

**Comment by xiywang** [comment]
@vivianQizhu @pezhang Updated. Please help review again, thanks.

**Comment by xiywang** [comment]
The case of nic hotplug/unplug is moved to another file, which will be submitted in another PR by wyu.
So just remove the case from this PR.

File: qemu/tests/memory_leak_after_nichotplug.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    logging.info("Guest free memory before nic hotplug: %d",
+                 free_mem_before_nichotplug)
+
+    if os_type == "windows" or params.get("hotplug_unplug_test", "no") == "yes":
+        error_context.context("Add network devices through monitor cmd",
+                              logging.info)
+        pci_model = params.get("pci_model")
+        netdst = params.get("netdst", "virbr0")
+        nettype = params.get("nettype", "bridge")
+        enable_msix_vectors = params.get("enable_msix_vectors")
```
Comment (vivianQizhu) [inline qemu/tests/memory_leak_after_nichotplug.py]: I don't think you need this parameter any more, you only have `no` for it.

File: qemu/tests/memory_leak_after_nichotplug.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    timeout = int(params.get("login_timeout", 360))
+    session = vm.wait_for_serial_login(timeout=timeout)
+    os_type = params.get("os_type")
+
+    free_mem_before_nichotplug = utils_misc.get_free_mem(session,
+                                                         os_type)
+    logging.info("Guest free memory before nic hotplug: %d",
+                 free_mem_before_nichotplug)
+
+    if os_type == "windows" or params.get("hotplug_unplug_test", "no") == "yes":
```
Comment (vivianQizhu) [inline qemu/tests/memory_leak_after_nichotplug.py]: Does not need `hotplug_unplug_test` either.

Comment (xiywang) [reply qemu/tests/memory_leak_after_nichotplug.py]: @vivianQizhu Updated.

**Comment by vivianQizhu** [comment]
@xiywang Code looks good to me, please provide latest test result, thx.

**Comment by xiywang** [comment]
Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.memory_leak_after_nichotplug: PASS (1206.46 s)
Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.x86_64.io-github-autotest-qemu.memory_leak_after_nichotplug: PASS (1076.27 s)

File: qemu/tests/cfg/memory_leak_after_nichotplug.cfg [hunk 4]
```diff
@@ -0,0 +1,5 @@
+- memory_leak_after_nichotplug:
+    type = memory_leak_after_nichotplug
+    pci_model = virtio-net-pci
+    enable_msix_vectors = no
```
Comment (vivianQizhu) [inline qemu/tests/cfg/memory_leak_after_nichotplug.cfg]: Sorry, one more tiny issue, this line could be removed right?

Comment (xiywang) [reply qemu/tests/cfg/memory_leak_after_nichotplug.cfg]: The default value of enable_msix_vectors is yes.
See guest-hw.cfg +37.
Anyway, it doesn't matter whether it's enabled or disabled in this case. So keep it or rm it are both ok I think.

Comment (vivianQizhu) [reply qemu/tests/cfg/memory_leak_after_nichotplug.cfg]: @xiywang I see, found the usage in command line generation, let's keep it as you wish.

**Pull request merged**